tools/deploy_ipk.sh

```diff
@@ -21,33 +27,35 @@ deploy() {
     DEST_FOLDER=/tmp/prplmesh_ipks
 
     echo "Removing previous ipks"
-    ssh "$TARGET" "rm -rf \"$DEST_FOLDER\" ; mkdir -p \"$DEST_FOLDER\""
+    eval ssh "$SSH_OPTIONS" "$TARGET" \""rm -rf $DEST_FOLDER ; mkdir -p $DEST_FOLDER\""
     echo "Copying $IPK to $TARGET:$DEST_FOLDER/$IPK_FILENAME"
-    scp "$IPK" "$TARGET:$DEST_FOLDER/$IPK_FILENAME"
+    eval scp "${SSH_OPTIONS} ${IPK} ${TARGET}:${DEST_FOLDER}/${IPK_FILENAME}"
 
-    ssh "$TARGET" <<EOF
+    eval ssh "$SSH_OPTIONS" "$TARGET" <<EOF
 # we don't want opkg to stay locked with a previous failed invocation.
 # when using this, make sure no one is using opkg in the meantime!
 pgrep opkg | xargs kill -SIGINT
 # remove any previously installed prplmesh:
 opkg remove prplmesh prplmesh-dwpal prplmesh-nl80211
 # currently opkg remove does not remove everything from /opt/prplmesh:
 rm -rf /opt/prplmesh
-opkg install -V2 "$DEST_FOLDER/$IPK_FILENAME"
+opkg install --force-reinstall -V2 "$DEST_FOLDER/$IPK_FILENAME"
 EOF
 
     if [ "$CERTIFICATION_MODE" = true ] ; then
         echo "Certification mode will be enabled on the target"
-        ssh "$TARGET" "uci set prplmesh.config.certification_mode=1 && uci commit"
+        eval ssh "$SSH_OPTIONS" "$TARGET" "uci set prplmesh.config.certification_mode=1 && uci commit"
         echo "Certification mode enabled on the target."
     fi
     echo "Done"
 }
 
 main() {
-        OPTS=$(getopt -o 'h' --long help,certification-mode -n 'parse-options' -- "$@")
-
-    if [ "$?" != 0 ] ; then err "Failed parsing options." >&2 ; usage; exit 1 ; fi
+    if ! OPTS=$(getopt -o 'h' --long help,certification-mode,proxy: -n 'parse-options' -- "$@"); then
+        err "Failed parsing options." >&2
```

Comment:
Only see it now, but if `err` always writes errors, couldn't it write to stderr itself? (it's a nitpick and not specific  to this PR of course).

Reply:
It certainly should, I'll try to remember to do that in a separate commit.

Reply:
actually err is undefined in this script, so I replaced it with echo.